src/authorization/AclOwnershipReader.ts

```diff
@@ -0,0 +1,52 @@
+import { CredentialGroup } from '../authentication/Credentials';
+import type { AccountSettings, AccountStore } from '../identity/interaction/email-password/storage/AccountStore';
+import type { AuxiliaryIdentifierStrategy } from '../ldp/auxiliary/AuxiliaryIdentifierStrategy';
+import type { AclPermission } from '../ldp/permissions/AclPermission';
+import type { PermissionSet } from '../ldp/permissions/Permissions';
+import type { PermissionReaderInput } from './PermissionReader';
+import { PermissionReader } from './PermissionReader';
+
+/**
+ * Allows full access on ACL resources if the request is being made by the owner of the pod containing that resource.
+ */
+export class AclOwnershipReader extends PermissionReader {
+  private readonly accountStore: AccountStore;
+  private readonly aclStrategy: AuxiliaryIdentifierStrategy;
+
+  public constructor(accountStore: AccountStore, aclStrategy: AuxiliaryIdentifierStrategy) {
+    super();
+    this.accountStore = accountStore;
+    this.aclStrategy = aclStrategy;
+  }
+
+  public async handle({ credentials, identifier }: PermissionReaderInput): Promise<PermissionSet> {
+    if (!this.aclStrategy.isAuxiliaryIdentifier(identifier)) {
```

Comment:
PS I assume this case is for performance reasons?  (If so, let's comment that.)
Because the `Control` access would still be given to non-ACL resources.

Reply:
Yes, otherwise there would be a call to the storage for every request, now it's just for every request targeting an acl resource.

Reply:
Check, let's just explain that in comment then.